### DISCUSSION TITLE
Provide Docker Images for download

### DISCUSSION THREAD

**Discussion by Archer57** [ORelio/Minecraft-Console-Client#1854] [Ideas]
Hi all,

I would love to see us providing Docker Images for our users.
For that we would have to extend our CI/CD Tools (ci.appveyor.com) or switch to GitHub Actions for Continuous Integration and add it there.

What is your opinion on this?

Best regards

**Comment by ReinforceZwei** [comment]
I don't use Docker very often and not familiar with it as well. I'm afraid that we don't have enough resource nor contributors to manage Docker Image. 

Regarding GitHub Actions, we have discussed before (see #1388 and #1267) and decided not to use it for now.

**Reply by Archer57** [reply]
That shouldn't be a problem. We can set up a pipeline to build Docker images by just running a docker build and pushing it to our package repository. I will describe that with some more detail in my reply to ORelio.

**Comment by ORelio** [comment]
We need a way to have development builds for anyone (not just GitHub users) to download. Feel free to investigate this, if you find a way to set up development builds that are easy to download for everyone, we'll switch to GitHub actions, which would be more convenient and secure than a third party service 👍

**Reply by Archer57** [reply]
I found 2 ways as we need one for classic .exe build results and one for Docker. Upfront, I think the official GitHub page on this gives a good impression on how easy this can be: https://blog.codecentric.de/en/2021/03/github-container-registry/

**1. Classic Builds (.exe)**
For classic builds, the most common way I've seen is to set up a classic CI/CD Pipeline and bind the build results on releases in GitHub. These build results can then be downloaded without login if you select the release. For reference, please have a look on [this repository](https://github.com/kennyvv/Alex). On the right side, you can see that they have several releases. If you select one of these, you can download a zip file for every platform you need.

**2. Docker**
For Docker I found [this](https://blog.codecentric.de/en/2021/03/github-container-registry/) article which gives a good introduction on how to publish Docker images with the GitHub Container Registry. The relevant topics are covered a bit after half the page under "3. Creating GitHub Actions workflow and logging into GitHub Container Registry using the PAT". If we take this one step apart and include the build itself to the workflow, we could do this automatically. After that is done, the users can just get the image by using `docker pull` (for his example `docker pull ghcr.io/jonashackt/hello-world`). This can be done without a GitHub account as well.

The advantage of releasing Docker Images on our own and storing them in one place is that if any user executes a `docker pull` on our project, he would by default get the latest version (easy for updating via script) or could as well specify a certain version he wants to have. Both ways I described can be used by anyone without a GitHub account.

**Reply by ORelio** [reply]
Thanks for pointing out to a repository that automatically uploads releases.
Maybe this [yml](https://github.com/kennyvv/Alex/blob/master/.github/workflows/dotnetcore.yml) file would help setting it up, however I've got limited time to investigate this.

**Reply by ORelio** [reply]
Thanks for your help, the repository you pointed to indeed helped me setting up automated releases through GitHub actions, combined with previous research in #1267. Docker building process is not updated yet (it was user-contributed).
